astropy/io/fits/card.py

```diff
@@ -194,8 +213,18 @@ def __init__(self, keyword=None, value=None, comment=None, **kwargs):
                 self.value = value
 
         if comment is not None:
+            # This sets the unit from the comment string (if any), but
+            # we prefer the unit in value, addressed next
```

Comment:
Ah!  So you actually Believe in Quantities :-).  As a newcomer, I am just coming around to the idea they might be useful and thus attempting to accommodate them.  With your help, I am learning a lot more about Quantities, warnings, etc. and starting to see how this can all fit together in a way that is low-impact to the community, but ads the requested feature in a convenient way.